tests/e2e/analyze_graph/optimize_indexes.py

```diff
@@ -278,5 +283,194 @@ def test_same_avg_group_size_diff_distribution(connect):
+    memgraph.execute("CREATE INDEX ON :SuperNode;")
+    memgraph.execute("CREATE INDEX ON :Node(id);")
+    memgraph.execute("CREATE INDEX ON :Node;")
+    memgraph.execute("CREATE INDEX ON :Node2(id);")
+    memgraph.execute("CREATE INDEX ON :Node2;")
+
+    memgraph.execute("match (n:Node) match (s:SuperNode {id: 1}) merge (n)<-[:HAS_REL_TO]-(s);")
+    memgraph.execute("match (n:Node2) match (s:SuperNode {id: 1}) merge (n)<-[:HAS_REL_TO]-(s);")
+
+    query = (
+        "explain match (n:Node) match (s:SuperNode {id: 1}) call { with n, s merge (n)<-[:HAS_REL_TO]-(s) } return 1"
+    )
+    expected_explain = [
+        f" * Produce {{0}}",
+        f" * Accumulate",
+        f" * Accumulate",
+        f" * Apply",
+        f" |\\ ",
+        f" | * EmptyResult",
+        f" | * Merge",
+        f" | |\\ On Match",
+        f" | | * Expand (s)-[anon3:HAS_REL_TO]->(n)",
+        f" | | * Once",
+        f" | |\\ On Create",
+        f" | | * CreateExpand (n)<-[anon3:HAS_REL_TO]-(s)",
+        f" | | * Once",
+        f" | * Produce {{n, s}}",
+        f" | * Once",
+        f" * ScanAllByLabel (n :Node)",
+        f" * ScanAllByLabelPropertyValue (s :SuperNode {{id}})",
+        f" * Once",
+    ]
+
+    result_without_analysis = list(memgraph.execute_and_fetch(query))
+    result_without_analysis = [x[OptimizeIndexesConstants.QUERY_PLAN] for x in result_without_analysis]
+    assert expected_explain == result_without_analysis
+
+    memgraph.execute("analyze graph;")
+
+    expected_explain = [
+        x.replace(f" | | * Expand (s)-[anon3:HAS_REL_TO]->(n)", f" | | * Expand (n)<-[anon3:HAS_REL_TO]-(s)")
+        for x in expected_explain
+    ]
+    result_with_analysis = list(memgraph.execute_and_fetch(query))
+    result_with_analysis = [x[OptimizeIndexesConstants.QUERY_PLAN] for x in result_with_analysis]
+
+    assert expected_explain == result_with_analysis
+
+
+def test_given_supernode_when_subquery_and_union_then_carry_information(memgraph):
+    memgraph.execute("FOREACH (i in range(1, 1000) | CREATE (:Node {id: i}));")
+    memgraph.execute("FOREACH (i in range(1, 1000) | CREATE (:Node2 {id: i}));")
+    memgraph.execute("CREATE (:SuperNode {id: 1});")
+    memgraph.execute("CREATE INDEX ON :SuperNode(id);")
+    memgraph.execute("CREATE INDEX ON :SuperNode;")
+    memgraph.execute("CREATE INDEX ON :Node(id);")
+    memgraph.execute("CREATE INDEX ON :Node;")
+    memgraph.execute("CREATE INDEX ON :Node2(id);")
+    memgraph.execute("CREATE INDEX ON :Node2;")
+
+    memgraph.execute("match (n:Node) match (s:SuperNode {id: 1}) merge (n)<-[:HAS_REL_TO]-(s);")
+    memgraph.execute("match (n:Node2) match (s:SuperNode {id: 1}) merge (n)<-[:HAS_REL_TO]-(s);")
+
+    query = "explain match (n:Node) match (s:SuperNode {id: 1}) call { with n, s merge (n)<-[:HAS_REL_TO]-(s) } return s union all match (n:Node) match (s:SuperNode {id: 1}) call { with n, s merge (n)<-[:HAS_REL_TO]-(s) } return s;"
```

Comment:
ok now I understand you want to pass this to the subquery. Crazy things going on :smile: